### ISSUE TITLE
[compile error] "Please add a short description here"

### ISSUE THREAD

**Issue by Amayess** [PointCloudLibrary/pcl#5272]
Hi,
Am actually creating an app using c++ and Qt. So far I have been able to display the camera stream on my application and apply the necessary post processing filters to get a better rendering and then generate my Cloud Point and export it. Now, I'm looking for a way to export the cloud point but in OBJ format or convert the PLY to OBJ directly, I tried to do that following this example: https://github.com/FZzzz/PlyToObj but I didn't get any result, in output I manage to generate an .obj but there is nothing inside, no mesh.

Can someone help me to solve it and understand it ?

**Comment by mvieth** [comment]
You can use `pcl::io::saveOBJFile` to write a .obj file, either from a [`pcl::PolygonMesh`](https://pointclouds.org/documentation/group__io.html#gabc0a54f1960ca03d5fa74fe3c63887a0) or a [`pcl::TextureMesh`](https://pointclouds.org/documentation/group__io.html#ga078c607bf51674d878775bacb11d34d6)

**Comment by Amayess** [comment]
Hi @mvieth 

I am actually trying to do it with your suggestion but i get a compilation error  **LNK1104: cannot open file ‘libboost_thread-vc142-mt-gd-x64-1_79.lib’** when i add **#include <pcl/io/obj_io.h>** to my project.

I tried with : #include <pcl/io/png_io.h> and it compiles correctly.

NB : I am using boost in this project and it worked with no problem, so in my opinion the problem is related to *obj_io.h", what do you think ? 

this is a screen of my code : 
![Capture](https://user-images.githubusercontent.com/105653860/170217921-ba3455c6-b80f-4b6a-9fd5-7736916b1dfa.JPG)

**Comment by mvieth** [comment]
My best guess would be that you just need to install that boost library?
Some other information would also be helpful:
- Which operating system do you use?
- Which PCL version do you use? How did you install it?
- Which compiler do you use? How do you build your own project?

**Comment by Amayess** [comment]
Hi @mvieth 

Boost is installed and compiled plus i already used it in this project in other cases, see image below.

I am using Windows 10 and visual studio 2019.

![boost](https://user
![boost 1](https://user-images.githubusercontent.com/105653860/170226454-aa17380a-1fee-44cd-8783-6c56a3bd5787.JPG)
![boost 1](https://user-images.githubusercontent.com/105653860/170226517-d46a2932-b68b-4c0d-b33a-432413445e3b.JPG)
![boost](https://user-images.githubusercontent.com/105653860/170226520-b8139954-ae4e-46a4-a2c9-8af41893972f.JPG)
-images.githubusercontent.com/105653860/170226443-98bf1f67-e84a-4ecf-85ac-fb92d152975c.JPG)

**Comment by Amayess** [comment]
Hi @mvieth

Boost is installed and compiled plus i already used it in this project in other cases, see image below.

I am using Windows 10 and visual studio 2019.
PCL version : 1.8.1
![boost 1](https://user-images.githubusercontent.com/105653860/170226780-5da7df70-fc3d-433c-93a0-3492019246c0.JPG)
![boost](https://user-images.githubusercontent.com/105653860/170226782-76303b98-4a6f-406c-ac87-4536d6623b2e.JPG)

**Comment by mvieth** [comment]
Can you check if the file `libboost_thread-vc142-mt-gd-x64-1_79.lib` exists on your computer?
Boost is actually a collection of several libraries, for example the thread library is one of them. Even when "Boost" is installed, that does not necessarily mean that _all_  of the Boost libraries are installed.
It would also be helpful if you could try a newer PCL version, e.g. the newest version 1.12.1. The PCL version 1.8.1 is quite old.

**Comment by Amayess** [comment]
Yes, the file exists inside the **stage** folder of the boost library.. and added in the linkers 
![lib](https://user-images.githubusercontent.com/105653860/170265601-e2710dfa-1280-4610-b838-2bffeb0787f7.JPG)
![lib1](https://user-images.githubusercontent.com/105653860/170265604-891fd417-456e-41a3-b0f6-a9990c9c6d8c.JPG)


![3](https://user-images.githubusercontent.com/105653860/170236340-cbf5f210-2003-4c68-a988-b05fb8117353.JPG)

**Comment by mvieth** [comment]
Then can you try again with PCL version 1.12.1?
PCL 1.8.1 is quite old and there have been many improvements and modernisations since then.

**Comment by Amayess** [comment]
Ok, will do that. 

I'll give you feedback.

**Comment by Amayess** [comment]
Ok, the first problem is resolved it was a path problem but since i installed the 1.12.1 version it compiles but when it's about to launch i get an error : **OpenN12.dll** not found.

And there is no **OpenN12.dll** inside the pcl 1.12.1 folder, where can i find it, please ?

**Comment by Amayess** [comment]
Update : fixed ! thank you for your help
